tools/api.py

```diff
@@ -323,47 +332,56 @@ def share_params(self):
                     size_type = Type(p[0])
             self.params.append(Parameter(size_param_name, size_type.str()))
 
-    def update(self):
+    def update(self) -> None:
         self.share_params()
         for param in self.params:
             param.update()
         if self.returns:
             self.returns.update()
         self.create_cfunction()
 
-    def inputs(self):
+    def inputs(self) -> str:
         return ', '.join([p.input() for p in self.params])
 
-    def input_map(self):
-        m = {}
+    # TODO: Shoule we remove Optional?
```

Comment:
I think it should be okay to have it `Optional` since they are function's input and return args.